src/plan.rs

```diff
@@ -126,31 +174,106 @@ impl<T: FFTnum> FftPlannerScalar<T> {
         if let Some(instance) = self.algorithm_cache.get(&len) {
             Arc::clone(instance)
         } else {
-            let instance = self.plan_new_fft_with_factors(len, PrimeFactors::compute(len));
-            self.algorithm_cache.insert(len, Arc::clone(&instance));
-            instance
+            let recipe = self.design_fft_for_len(len);
+            let fft = self.build_fft(recipe);
+            self.algorithm_cache.insert(len, Arc::clone(&fft));
+            fft
         }
     }
 
-    fn plan_fft_with_factors(&mut self, len: usize, factors: PrimeFactors) -> Arc<dyn Fft<T>> {
-        if let Some(instance) = self.algorithm_cache.get(&len) {
-            Arc::clone(instance)
+    // Make a recipe for a length
+    fn design_fft_for_len(&mut self, len: usize) -> Recipe {
+        if len < 2 {
+            Recipe::DFT(len)
+        } else {
+            let factors = PrimeFactors::compute(len);
+            self.design_fft_with_factors(len, factors)
+        }
+    }
+
+    // Create the fft from a recipe
+    fn build_fft(&mut self, plan: Recipe) -> Arc<dyn Fft<T>> {
```

Comment:
One problem I notice with this `build_fft` is that it doesn't respect cache. For example, if the recipe is a MixedRadix, it will recursively call `build_fft` and construct new instances of the children, even if they already exist in cache.

Reply:
This should be taken care of now!